spec/discount_spec.rb

```diff
@@ -0,0 +1,56 @@
+require_relative '../app/lib/discount'
+
+describe 'discounts' do
+
+  let(:dummy_receipt) { Class.new { extend Discounts } }
+
+  describe 'calculating a discount amount' do
+
+    it 'should calculate a discount based on amount to be discounted and discount rate' do
+      expect(dummy_receipt.calc_discount(85, 10)).to eq(8.5)
+    end
+
+  end
+
+  describe 'calculating a 5% discount when net total above 50' do
+
+    it 'should calculate a discount of 5% on an order with a pre-tax total over $50' do
+      allow(dummy_receipt).to receive(:net_total).and_return(55)
+      expect(dummy_receipt.discount_total_price).to eq(2.75)
+    end
+
+    it 'should not offer a discount on an order where pre-tax total is less than $50' do
+      allow(dummy_receipt).to receive(:net_total).and_return(49)
+      expect(dummy_receipt.discount_total_price).to eq(0)
+    end
+
+  end
+
+  describe 'calculating a discount of 10% on all mufins' do
+
+    it 'should offer a discount of 10% on the total spend on muffins' do
+      allow(dummy_receipt).to receive(:line_price).with({:item=>"Blueberry Muffin", :quantity=>2}).and_return(8.10)
```

Comment:
Metrics/LineLength: Line is too long. [115/80]<br>Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.<br>Style/HashSyntax: Use the new Ruby 1.9 hash syntax.